src/material/datepicker/calendar.ts

```diff
@@ -393,7 +418,19 @@ export class MatCalendar<D> implements AfterContentInit, AfterViewChecked, OnDes
   }
 
   ngOnChanges(changes: SimpleChanges) {
-    const change = changes['minDate'] || changes['maxDate'] || changes['dateFilter'];
+    const minDateChange: SimpleChange | undefined = !this._dateAdapter.sameDate(
+      this._minDate,
+      this._oldMinDate,
```

Comment:
Do _oldMinDate and _oldMaxDate exist for the sole purpose of comparing them here? If so, SimpleChange has a reference to the previous value you can use - https://angular.io/api/core/SimpleChange

Reply:
done ✔️